daemon/logic/engine.go

```diff
@@ -479,3 +479,14 @@ func (e *Engine) GenerateKeypair(ctx context.Context, notifier *observer.Notifie
 
 	return nil
 }
+
+// ChangePassword returns the new password object and re-encrypted masterkey object
+func (e *Engine) ChangePassword(ctx context.Context, newPassword string) (*primitive.UserPassword, *primitive.MasterKey, error) {
+	// We need to re-use the master key
+	currentMasterKey, err := e.crypto.UnsealMasterKey(ctx)
```

Comment:
The master key shouldn't be extractable from the crypto package in its unencrypted form, to keep it easy to track.

Instead, how about a `crypto.Engine.ChangePassword(ctx, oldpass, newpass)` method?